graylog2-server/src/main/java/org/graylog2/telemetry/cluster/TelemetryClusterInfoPeriodical.java

```diff
@@ -0,0 +1,124 @@
+    private final ServerStatus serverStatus;
+    private final String clusterId;
+    private final LeaderElectionService leaderElectionService;
+    private final DBTelemetryClusterInfo dbTelemetryClusterInfo;
+    private final Duration telemetryClusterInfoTtl;
+
+    @Inject
+    public TelemetryClusterInfoPeriodical(@Named(TELEMETRY_CLUSTER_INFO_TTL) Duration telemetryClusterInfoTtl,
+                                          ServerStatus serverStatus,
+                                          ClusterConfigService clusterConfigService,
+                                          LeaderElectionService leaderElectionService,
+                                          DBTelemetryClusterInfo dbTelemetryClusterInfo) {
+        this.telemetryClusterInfoTtl = telemetryClusterInfoTtl;
+        this.serverStatus = serverStatus;
+        this.clusterId = Optional.ofNullable(clusterConfigService.get(ClusterId.class))
+                .map(ClusterId::clusterId)
+                .orElse(null);
+        this.leaderElectionService = leaderElectionService;
+        this.dbTelemetryClusterInfo = dbTelemetryClusterInfo;
+    }
+
+    @Override
+    public void doRun() {
+        Map<String, Object> nodeInfo = new ImmutableMap.Builder<String, Object>()
+                .put("facility", "graylog-server")
+                .put("codename", ServerVersion.CODENAME)
+                .put(FIELD_NODE_ID, serverStatus.getNodeId().toString())
+                .put("cluster_id", clusterId)
+                .put("version", ServerVersion.VERSION.toString())
+                .put("started_at", Tools.getISO8601String(serverStatus.getStartedAt()))
+                .put("hostname", Tools.getLocalCanonicalHostname())
+                .put("lifecycle", serverStatus.getLifecycle().getDescription().toLowerCase(Locale.ENGLISH))
+                .put("lb_status", serverStatus.getLifecycle().getLoadbalancerStatus().toString().toLowerCase(Locale.ENGLISH))
+                .put("timezone", serverStatus.getTimezone().getID())
+                .put("operating_system", System.getProperty("os.name", "unknown") + " " + System.getProperty("os.version", "unknown"))
+                .put("is_leader", leaderElectionService.isLeader())
+                .put("is_processing", serverStatus.isProcessing())
+                .build();
+
+        dbTelemetryClusterInfo.update(nodeInfo, serverStatus.getNodeId().toString());
+
+    }
+
+    @Override
+    public boolean runsForever() {
+        return false;
+    }
+
+    @Override
+    public boolean stopOnGracefulShutdown() {
+        return true;
+    }
+
+    @Override
+    public boolean startOnThisNode() {
+        return true;
+    }
+
+    @Override
+    public boolean isDaemon() {
+        return true;
+    }
+
+    @Override
+    public int getInitialDelaySeconds() {
+        return 0;
+    }
+
+    @Override
+    public int getPeriodSeconds() {
+        return Ints.saturatedCast(telemetryClusterInfoTtl.minusMinutes(1).toSeconds());
+    }
```

Comment:
We create a mongodb ttl index depending on the telemetry_cluster_info_ttl configuration property. Before the document is removed due to the ttl index, live nodes update this document one minute earlier. For dead nodes, the document is removed because of the ttl index. The one minute is just to make sure that the node has enough time to update the document.